substrate/client/network/src/peer_info.rs

```diff
@@ -71,6 +72,8 @@ pub struct PeerInfoBehaviour {
 	garbage_collect: Pin<Box<dyn Stream<Item = ()> + Send>>,
 	/// Record keeping of external addresses. Data is queried by the `NetworkService`.
 	external_addresses: ExternalAddresses,
+	/// Whether only these addresses should be advertised to remote peers in [`Identify`] messages.
+	public_addresses_only: Option<Vec<Multiaddr>>,
```

Comment:
Do we need to store the public addresses in this object?

When we construct the object we trigger an `Identify::NewExternalAddr` for each of those addresses.
Then, we only need to know if further external addresses from the swarm need propagation to the `identify`.
Could we use a flag here instead?

Reply:
Done